fetch/cosmwasm_contract/src/state.rs

```diff
@@ -1,23 +1,21 @@
 use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
 
-use cosmwasm_std::{CanonicalAddr, HumanAddr, Storage};
-use cosmwasm_storage::{singleton, singleton_read, ReadonlySingleton, Singleton, PrefixedStorage, ReadonlyPrefixedStorage};
-
 use crate::msg::Uint128;
+use cosmwasm_std::{HumanAddr, ReadonlyStorage, Storage};
+use cosmwasm_storage::{
+    singleton, singleton_read, PrefixedStorage, ReadonlyPrefixedStorage, ReadonlySingleton,
+    Singleton,
+};
 
 pub static CONFIG_KEY: &[u8] = b"config";
-pub static ACCESS_CONTROL_KEY: &[u8] = b"access_control";
-
-// FIXME(LR) use these instead of numbers
-//pub const DELEGATE_ROLE: &[u8] = b"DELEGATE_ROLE";
-//pub const RELAYER_ROLE: &[u8] = b"RELAYER_ROLE";
+pub static REFUNDS_KEY: &[u8] = b"refunds";
 
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
 #[serde(rename_all = "snake_case")]
 pub struct State {
-    pub supply: Uint128,               // amount of token migrated to the other chain
-    pub refunds_fees_accrued: Uint128, // fees used to pay for refunds
+    pub supply: Uint128,       // amount of token migrated to the other chain
+    pub fees_accrued: Uint128, // fees
```

Comment:
You might consider to drop the `fees_accrued` state variable. You can derive its value from `= contract.balance - supply`.
This is just to spare gas for storing/reading additional state variable and +/- operations on this additional variable.

The downside is that then we would NOT be unable to distinguish between fees collected and wild unsolicited token transfers sent directly to address of our contract. This is excatly the design decission made on Ethereum contract side.

Reply:
for the downside reason you cited, I decided to have them in a separate state variable.
we can discuss it more during our call